onnx2kerastl/reshape_layers.py

```diff
@@ -202,20 +202,31 @@ def target_layer(x):
 
             else:
                 input_0 = ensure_tf_type(layers[node.input[0]], name="%s_const" % keras_name)
-                logger.debug('The first argument is Keras/tf layer. Apply keras.Reshape.')
-                logger.debug('Target shape :')
-                logger.debug(np.int32(input_1[1:]))
-
-                if len(np.int32(input_1[1:])) == 1 and np.int32(input_1[1:])[0] == -1:
-                    logger.debug('The first argument is Keras/tf layer. Apply keras.Flatten.')
-                    flatten = keras.layers.Flatten(name=keras_name)
-                    layers[node_name] = flatten(input_0)
+                input_0_shape = input_0.shape
+                first_mismatch = np.argmin(np.array(input_0_shape[:len(input_1)]) == input_1)
+                if (input_1 == None).any() and (np.array(input_0_shape) == None).any() and len(input_1) < len(input_0_shape)\
+                        and input_1[first_mismatch] == -1: #reshape end
+                    end_match_arr = np.array(input_0_shape[-len(input_1):]) == input_1
+                    end_idx_match = np.argmax((np.array(input_0_shape[-len(input_1):]) == input_1))
+                    end_idx_match = end_idx_match + len(input_0_shape) - len(input_1) if end_idx_match > first_mismatch \
+                                                     and end_match_arr[end_idx_match] else len(input_0_shape) + 1
+                    tf_shape = tf.shape(input_0)
+                    layers[node_name] = tf.reshape(input_0, [*tf_shape[:first_mismatch], -1, *tf_shape[end_idx_match:]])
```

Comment:
replace with `tf.reshape(input_0, input1)`

Reply:
Discussed this together (won't work in this way since input1 should be a kerastensor).